src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -313,9 +352,9 @@ def __getitem__(self, key: SupportsIndex | datetime | slice) -> float | ArrayLik
                     key = slice(slice(key.start, key.stop).indices(self.__len__()))
             elif isinstance(key.start, datetime) or isinstance(key.stop, datetime):
                 if key.start is None:
-                    key = slice(self._buffer.time_bound_oldest, key.stop)
+                    key = slice(self._buffer.oldest_timestamp, key.stop)
```

Comment:
you can remove the `._buffer` now ;)

Reply:
Didn't want to touch it since I am reworking this functionality at the moment for #214 where I remove this part. Also they are not exactly the same as they differ for empty arrays (which is not a concern here due to the check couple of lines above).